src/lib/pubkey/classic_mceliece/cmce_encaps.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Classic McEliece Encapsulation
+ * Based on the public domain reference implementation by the designers
+ * (https://classic.mceliece.org/impl.html - released in Oct 2022 for NISTPQC-R4)
+ *
+ * (C) 2023 Jack Lloyd
+ *     2023,2024 Fabian Albert, Amos Treiber - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ **/
+#include <botan/internal/cmce_encaps.h>
+
+#include <botan/rng.h>
+#include <iostream>
+
+namespace Botan {
+
+bitvector Classic_McEliece_Encryptor::encode(const Classic_McEliece_Parameters& params,
+                                             const secure_bitvector& e,
+                                             const Classic_McEliece_Matrix& mat) {
+   return mat.mul(params, e);
+}
+
+std::optional<secure_bitvector> Classic_McEliece_Encryptor::fixed_weight_vector_gen(
+   const Classic_McEliece_Parameters& params, const secure_vector<uint8_t>& rand) {
+   BOTAN_ASSERT_NOMSG(rand.size() == params.tau() * (params.sigma1() / 8));
```

Comment:
Instead of double-checking the amount of randomness, you might also just pass the RNG in here and then pull as many bytes as you need.